### PR TITLE
Remove .Net Core 1.0 requirement for Loc and fix mac build

### PR DESCRIPTION
**Pull request by kevcunnane**
Use StringResourceTool to remove .Net Core 1.0 dependency for Loc
Fixed Mac build by copying the code used in internal repo for build.
- This now mostly matches internal, so expect this should work OK.

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -554,12 +554,24 @@ Task("SetPackageVersions")
 Task("SRGen")
 	.Does(() =>
 {
+    // save current working directory to restore at end of task
+    var taskStartedWorkingDirectory = System.IO.Directory.GetCurrentDirectory();
+    var relativePathPrefix = "." + System.IO.Path.DirectorySeparatorChar.ToString();
```
Comment (Charles-Gagnon) [inline build.cake]: You shouldn't need to add the separator char here - just using "." in Path.Combine should work fine.

File: build.cake [hunk 2]
```diff
@@ -554,12 +554,24 @@ Task("SetPackageVersions")
...
+
     var projects = System.IO.Directory.GetFiles(sourceFolder, "*.csproj", SearchOption.AllDirectories).ToList();
     var locTemplateDir = System.IO.Path.Combine(sourceFolder, "../localization");
```
Comment (Charles-Gagnon) [inline build.cake]: Hm, did you verify this works on macs? Anyways might be good to fix it up and just make it

```
Path.Combine(sourceFolder, "..", "localization");
```

File: build.cake [hunk 3]
```diff
@@ -635,6 +648,9 @@ Task("SRGen")
             templateDoc.SaveAsResX(newPath.Replace("xlf","resx"));        
         }
     }
+
+    // restore the original working directory
+    System.IO.Directory.SetCurrentDirectory(taskStartedWorkingDirectory);
```
Comment (Charles-Gagnon) [inline build.cake]: Put this in a try/finally

File: build.cake (hunk 1 above)
Comment (kevcunnane) [reply build.cake]: Good point - will fix. FYI I'm porting this over from @kenvanhyning 's fix in other repo, so we should likely update there too.

File: build.cake (hunk 2 above)
Comment (kevcunnane) [reply build.cake]: It's a mac-related fix 😄 so definitely tested there!

File: build.cake (hunk 3 above)
Comment (kevcunnane) [reply build.cake]: Will do

**Pull request merged**